flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicSource.java

```diff
@@ -150,7 +166,9 @@ public KafkaDynamicSource(
             StartupMode startupMode,
             Map<KafkaTopicPartition, Long> specificStartupOffsets,
             long startupTimestampMillis,
-            boolean upsertMode) {
+            boolean upsertMode,
+            ObjectIdentifier objectIdentifier,
```

Comment:
nit: instead of passing catalog objects into the source, how about you already create an `String sourceName` in the factory, this avoid calling `asSummaryString` at multiple locations